enhancements/cloud-integration/tokenized-auth-enablement-operators-on-cloud.md

```diff
@@ -0,0 +1,404 @@
+specifically HyperShift/Rosa adding CCO with these token-aware changes or continuing with the current support for the 
+pod identity webhook.
+
+CCO's token-aware mode will work by adding, for example for STS, a "role-arn"-type field on operator added 
+CredentialsRequest objects. This is a new API field on the CredentialsRequest spec. When CCO acquires a 
+CredentialsRequest it will create a Secret (as today mostly) with new functionality: Adding to the Secret a path to the
+projected ServiceAccount token (static for OCP) and a roleARN (from the new field on the CredentialsRequest).
+
+CCO with the changes anticipated here, operating in Manual mode, will continue to ignore CredentialsRequests that do not
+contain the new STS related fields and as these are new fields, existing clusters won't have CredentialsRequests with 
+those fields already.
+
+Validation of CredentialsRequest by this new token-aware CCO? Maybe. Currently, CCO gets some permissions under ROSA
+(unused), could expand these permissions to include tools like AWS's Policy Simulator such that it could validate a role
+has permissions, and with this CCO could be the alerting mechanism for a changed CredentialsRequest without sufficient 
+permissions.
+
+CCO should consider adding the above, three-part test as a utility function exposed in the API.
+
+**HyperShift changes**: Include Cloud Credential Operator with token-aware mode (see above). Allows for processing of 
+CredentialsRequest objects added by operators.
+
+**OperatorHub and Console changes**: Allow for input from user of additional fields during install depending on the 
+Cloud Provider which will result in ENV variables on the Subscription object that are required in the 
+CredentialsRequests created by the Operator. Setting the Subscription config.ENV will allow UX for the information 
+needed by CCO or webhook, for input of the cloud credentials while not having to change the Subscription API.
+CLI Users will need to add the following to their subscription:
+
+```yaml
+kind: Subscription
+metadata:
+ name: ...
+spec:
+  config:
+    env:
+    - name: ROLEARN
+      value: "<role ARN >"
+```
+
+Show in OperatorHub that the cluster is in a mode that supports token-based authentication by reading the 
+`.spec.serviceAccountIssuer` from the Authentication CR, `.status.platformStatus.type` from the Infrastructure CR,
+`.spec.credentialsMode` from the CloudCredentials CR.
+
+Show that the operator is enabled for Token-based use by reading the CSV annotation provided by the operator author.
+
+Generate a manual ack for operator upgrade when cloud resources are involved. This can be determined by parsing the 
+Subscription objects. Ensures admin signs off that cloud resources are provisioned and working as intended before an 
+operator upgrade. This could be by setting the default subscription mode to Manual in the UI for operators that support 
+TAT.
```

Comment:
we're at the point in the EP where we need to start making choices.  So state which one we're actually going to implement, and if you think we'll do something else in the future, then include that as "future iterations could..."

i'd be inclined to just default to manual approvals for now (though really i think we should have always been doing that for all operator installs).

I think we need more data on how often operator permissions actually change before we introduce a mandatory admin-ack for upgrading.  We also need to see/think about whether we can do any better (i.e. proactively detecting when permissions have changed and only requiring the ack in that situation).